qiskit_machine_learning/algorithms/inference/qbayesian.py

```diff
@@ -0,0 +1,331 @@
+                by :class:`~qiskit.primitives.Sampler` will be used.
+        Raises:
+            ValueError: If any register in the circuit is not mapped to exactly one qubit.
+        """
+        # Test valid input
+        for qrg in circuit.qregs:
+            if qrg.size > 1:
+                raise ValueError("Every register needs to be mapped to exactly one unique qubit")
+        # Initialize parameter
+        self._circ = circuit
+        self.limit = limit
+        self.threshold = threshold
+        if sampler is None:
+            sampler = Sampler()
+        self.sampler = sampler
+
+        # Label of register mapped to its qubit
+        self._label2qubit = {qrg.name: qrg[0] for qrg in self._circ.qregs}
+        # Label of register mapped to its qubit index bottom up in significance
+        self._label2qidx = {
+            qrg.name: self._circ.num_qubits - idx - 1 for idx, qrg in enumerate(self._circ.qregs)
+        }
+        # Distribution of samples from rejection sampling
+        self.samples: Dict[str, float] = {}
+        # True if rejection sampling converged after limit
+        self.converged = bool()
+
+    def _get_grover_op(self, evidence: Dict[str, int]) -> GroverOperator:
+        """
+        Constructs a Grover operator based on the provided evidence. The evidence is used to
+        determine the "good states" that the Grover operator will amplify.
+
+        Args:
+            evidence: A dictionary representing the evidence with keys as variable labels
+                and values as states.
+        Returns:
+            GroverOperator: The constructed Grover operator.
+        """
+        # Evidence to reversed qubit index sorted by index
+        num_qubits = self._circ.num_qubits
+        e2idx = sorted(
+            [(self._label2qidx[e_key], e_val) for e_key, e_val in evidence.items()],
+            key=lambda x: x[0],
+        )
+        # Binary format of good states
+        num_evd = len(e2idx)
+        bin_str = [
+            format(i, f"0{(num_qubits - num_evd)}b") for i in range(2 ** (num_qubits - num_evd))
+        ]
+        # Get good states
+        good_states = []
+        for b in bin_str:
+            for e_idx, e_val in e2idx:
+                b = b[:e_idx] + str(e_val) + b[e_idx:]
+            good_states.append(b)
+        # Get statevector by transform good states w.r.t its index to 1 and o/w to 0
+        oracle = Statevector(
+            [int(format(i, f"0{num_qubits}b") in good_states) for i in range(2**num_qubits)]
+        )
+        return GroverOperator(oracle, state_preparation=self._circ)
+
+    def _run_circuit(self, circuit: QuantumCircuit) -> Dict[str, float]:
+        """Run the quantum circuit with the sampler."""
+        # Sample from circuit
+        job = self.sampler.run(circuit)
+        result = job.result()
+        # Get the counts of quantum state results
+        counts = result.quasi_dists[0].binary_probabilities()
```

Comment:
In the case of noise mitigation (M3) being applied the distribution can have negative values (the total probability is still one). I imagine that does not matter for the usage here but just checking with you to make sure. There is a method on the quasi probability to return a normalized version if it does.

Reply:
No, it is good that you mention it. There was code for it, which I removed to make the file cleaner. Were you referring to this method: qiskit.result.QuasiDistribution.nearest_probability_distribution()?

Reply:
Yes, that seems to be the method. If negative values are ok then its fine.

Reply:
I will change it. Thank you for your constructive feedback. I hope the changes have been made as intended.